File: fullsimpe/SimPe ExamplePlugin/MyWindowPlugin.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SimPe.Plugin
{
    /// <summary>
    /// An instance of this class will hook into SimPEs Tools Menu, allowing the users to start your Plugin
    /// </summary>
    public class MyWindowPlugin : SimPe.Interfaces.IToolPlus
    {
        #region IToolPlus Member

        public void Execute(object sender, SimPe.Events.ResourceEventArgs e)
        {
            if (!ChangeEnabledStateEventHandler(sender, e)) return;


            MyWindowPluginForm f = new MyWindowPluginForm();
            ///
            /// TODO: Do some initialization work with your Form
            /// 
            f.ShowDialog();
            ///
            /// TODO: Do some finalization work with your Form
            /// 
            f.Dispose();
        }

        public bool ChangeEnabledStateEventHandler(object sender, SimPe.Events.ResourceEventArgs e)
        {
            ///
            /// TODO: return true here, if the Plugin should be enabled, and false otherwise
            /// e contains a List of Resources selected in the SimPE GUI as well as the currently 
            /// loaded package
            /// 
            return true;
        }

        #endregion

        #region IToolExt Member

        public System.Drawing.Image Icon
        {
            get { 
                ///
                /// TODO: You can return an Image here, that will be displayed as the ICon for your Plugin in the SimPE Plugin Menu.
                /// 
                return null; 
            }
        }

        public System.Windows.Forms.Shortcut Shortcut
        {
            get {
                ///
                /// TODO: Return the Shurtcut key, that will start your Plugin
                /// 
                return System.Windows.Forms.Shortcut.None;
            }
        }

        public bool Visible
        {
            get { 
                ///
                /// TODO: Return false here, if you donÄt want your Plugin to show up in the Plugins Menus, but still want to catch the 
                /// ChangeEnabledStateEventHandler Event
                /// 
                return true; 
            }
        }

        #endregion

        public override string ToString()
        {
            ///
            /// TODO: Return the Name of your Plugin here
            /// 
            return "Object Creation\\Example Plugin";
        }
    }
}

File: fullsimpe/SimPe ExamplePlugin/VB Project/SimPe.Plugin/MyWindowPlugin.vb
Imports SimPe.Events
Imports SimPe.Interfaces
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Namespace SimPe.Plugin
    Public Class MyWindowPlugin
        Implements IToolPlus, IToolExt, IToolPlugin

        ' Methods
        Public Function ChangeEnabledStateEventHandler(ByVal sender As Object, ByVal e As ResourceEventArgs) As Boolean Implements IToolPlus.ChangeEnabledStateEventHandler
            Return True
        End Function

        Public Sub Execute(ByVal sender As Object, ByVal e As ResourceEventArgs) Implements IToolPlus.Execute
            If Me.ChangeEnabledStateEventHandler(sender, e) Then
                Dim form1 As New MyWindowPluginForm
                form1.ShowDialog()
                form1.Dispose()
            End If
        End Sub

        Public Overrides Function ToString() As String Implements IToolPlugin.ToString
            Return "Object Creation\Example Plugin"
        End Function


        ' Properties
        Public ReadOnly Property Icon() As Image Implements IToolExt.Icon
            Get
                Return Nothing
            End Get
        End Property

        Public ReadOnly Property Shortcut() As Shortcut Implements IToolExt.Shortcut
            Get
                Return Shortcut.None
            End Get
        End Property

        Public ReadOnly Property Visible() As Boolean Implements IToolExt.Visible
            Get
                Return True
            End Get
        End Property

    End Class
End Namespace


